lib/modules/manager/nix/artifacts.ts

```diff
@@ -0,0 +1,81 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { readLocalFile } from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  config,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  const lockFileName = packageFileName.replace(regEx(/\.nix$/), '.lock');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    logger.debug('No flake.lock found');
+    return null;
+  }
+
+  let cmd: string;
+
+  if (config.isLockFileMaintenance) {
+    cmd = `nix \
+    --extra-experimental-features nix-command \
+    --extra-experimental-features flakes \
+    flake update`;
+  } else {
+    const inputs = updatedDeps
+      .map(({ depName }) => depName)
+      .filter(is.nonEmptyStringAndNotWhitespace)
+      .map((depName) => `--update-input ${depName}`)
+      .join(' ');
+    cmd = `nix \
+    --extra-experimental-features nix-command \
+    --extra-experimental-features flakes \
+    flake lock ${inputs}`;
+  }
+  const execOptions: ExecOptions = {
+    cwdFile: packageFileName,
+    toolConstraints: [
+      {
+        toolName: 'nix',
+        constraint: config.constraints?.nix,
```

Comment:
not possible to read from repo files?

Reply:
The closest thing I can think of is the lock file version: https://github.com/JamieMagee/renovate-flake/blob/1a002f077cb4e727b324fad9b8c1518009db5eb8/test-1/flake.lock#L26

I'll see if there's a definite mapping from lock file version to nix version

Reply:
I can't find a mapping documented anywhere, except in code.

References:
- https://github.com/NixOS/nix/blob/9550b1d51933a51fbb21563db0e3f53d0e8faea8/src/libexpr/flake/lockfile.cc#L174
- https://github.com/NixOS/nix/blob/9550b1d51933a51fbb21563db0e3f53d0e8faea8/src/libexpr/flake/lockfile.cc#L71-L73

Reply:
All latest patch versions of nix from `2.4` onwards (when lockfiles were introduced` have `version` set to `7`.

Reply:
I don't think it's possible to resolve this with the information that we have available at runtime.

Reply:
ok,we can improve later, but we should add support for known lockfile versions soon to not update the lockfile from renovate update PR's